### PR TITLE
tests: added data channel test

### PR DESCRIPTION
**Pull request by Rugvip**
The first test is currently failing, most likely because the SCTP association isn't set up when we try to send data.

The last 3 should pass though.

### CODE REVIEW

**Comment by Rugvip** [comment]
Verified on OS X and made sure it doesn't break Android and iOS builds. Would be great if someone could test it on Linux.

File: tests/test_data_channel.c [hunk 1]
```diff
@@ -0,0 +1,280 @@
+/*
+ * Copyright (c) 2014, Ericsson AB. All rights reserved.
```
Comment (stefanalund) [inline tests/test_data_channel.c]: 2015

**Comment by ford-prefect** [comment]
Work on Linux (with the first test failing).

Might be nice to have this use libcheck if it's available on OS X already.

**Comment by superdump** [comment]
libcheck is supposed to work on os x. We could add a recipe for building it from source and start using it. Or whatever is appropriate.

http://check.sourceforge.net/web/install.html

**Comment by Rugvip** [comment]
Updated, the first test isn't expected to fail anymore :P, it's now waiting for the first data channel to be opened.

**Comment by superdump** [comment]
Merged.

**Pull request closed without merging**